internal/charmstore/store_test.go

```diff
@@ -1814,3 +1814,258 @@ func (s *StoreSuite) TestUpdateBaseEntity(c *gc.C) {
 		}
 	}
 }
+
+var promulgateTests = []struct {
```

Comment:
Great example of a table-driven test that's so much easier to understand than if it had been written otherwise.